roles/test_operator/tasks/main.yml

```diff
@@ -0,0 +1,248 @@
+  when: not cifmw_test_operator_dry_run | bool
+
+- name: Get full name of the test-operator csv
+  ansible.builtin.set_fact:
+    test_operator_csv_name: >-
+      {{ csv_list | json_query('resources[*].metadata.name') | select('match', 'test-operator') | first }}
+  when: not cifmw_test_operator_dry_run | bool
+
+- name: Wait for the test-operator csv to Succeed
+  kubernetes.core.k8s_info:
+    kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
+    api_key: "{{ cifmw_openshift_token | default(omit)}}"
+    context: "{{ cifmw_openshift_context | default(omit) }}"
+    namespace: "{{ cifmw_test_operator_namespace }}"
+    kind: ClusterServiceVersion
+    api_version: operators.coreos.com/v1alpha1
+    name: "{{ test_operator_csv_name }}"
+  register: csv
+  retries: 20
+  delay: 10
+  until: csv.resources[0].status.phase | default(omit) == "Succeeded"
+  when: not cifmw_test_operator_dry_run | bool
+
+- name: Start Tempest tests
+  kubernetes.core.k8s:
+    kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
+    api_key: "{{ cifmw_openshift_token | default(omit)}}"
+    context: "{{ cifmw_openshift_context | default(omit)}}"
+    state: present
+    wait: true
+    definition: "{{ cifmw_test_operator_tempest_config }}"
+  when: not cifmw_test_operator_dry_run | bool
+
+- name: Wait for the Tempest job to be Ready
+  kubernetes.core.k8s_info:
+    kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
+    api_key: "{{ cifmw_openshift_token | default(omit) }}"
+    context: "{{ cifmw_openshift_context | default(omit) }}"
+    namespace: "{{ cifmw_test_operator_namespace }}"
+    kind: Job
+    name: "{{ cifmw_test_operator_tempest_name }}"
+    wait: true
+  when: not cifmw_test_operator_dry_run | bool
+
+- name: Wait for the Tempest job to be Completed
+  kubernetes.core.k8s_info:
+    kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
+    api_key: "{{ cifmw_openshift_token | default(omit) }}"
+    context: "{{ cifmw_openshift_context | default(omit) }}"
+    namespace: "{{ cifmw_test_operator_namespace }}"
+    kind: Job
+    name: "{{ cifmw_test_operator_tempest_name }}"
+    wait: true
+    wait_timeout: "{{ cifmw_test_operator_timeout }}"
+    wait_condition:
+      type: Complete
+      status: true
+  when: not cifmw_test_operator_dry_run | bool
+
+- name: Start test-operator-logs-pod
+  when: not cifmw_test_operator_dry_run | bool
+  kubernetes.core.k8s:
+    kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
+    api_key: "{{ cifmw_openshift_token | default(omit)}}"
+    context: "{{ cifmw_openshift_context | default(omit)}}"
+    state: present
+    definition:
+      apiVersion: v1
+      kind: Pod
+      metadata:
+        name: test-operator-logs-pod
+        namespace: "{{ cifmw_test_operator_namespace }}"
+      spec:
+        containers:
+          - name: test-operator-logs-container
+            image: "{{ cifmw_test_operator_logs_image }}"
+            command: ["/bin/sh", "-c", "--"]
```

Comment:
in the command you typically specify only the arg0  "/bin/sh", and the rest of the args  go to args.

busybox sleep infinite # works too , no need for loop.

you might want to use oc run  with overrides:
https://gist.github.com/welshstew/4f9532fd8737de1a43277c9154caa95e

You can specify a container which  oc run --overrides tar -c mydir | tar -x  .
In this case you can run a container which runs tar and have the pv mount and just getting the output stream in single command.

Reply:
I updated the code so that I use the sleep infinity instead and fixed the `command:` and `args:` sections.

I will take a look at the ` oc run --overrides tar -c mydir | tar -x`. At this point, I just want to get this role merged and this can be addressed without breaking the backward compatibility when I will be incorporating tobiko to the role. I hope it is ok.